pysd/py_backend/output.py

```diff
@@ -0,0 +1,611 @@
+    """
+    def __init__(self, next):
+        super().__init__(next)
+        self.ds = None
+        self.out_file = None
+
+    def process_output(self, out_file):
+        """
+        If this handler can process out_file, it returns True, else False.
+        DataFrameHandler handles outputs to be saved as *.csv or *.tab files,
+        and is the default handler when no output file is passed by the user.
+
+        Parameters
+        ----------
+        out_file: str or pathlib.Path
+            Path to the file where the results will be written.
+
+        Returns
+        -------
+        None or DataFrameHandler instance
+
+        """
+        self.out_file = out_file
+
+        if not out_file:
+            return self
+
+        if out_file.suffix in [".csv", ".tab"]:
+            return self
+
+    def initialize(self, model, capture_elements):
+        """
+        Creates a pandas DataFrame and adds model variables as columns.
+
+        Parameters
+        ----------
+        model: pysd.Model
+            PySD Model object
+        capture_elements: set
+            Which model elements to capture - uses pysafe names.
+
+        Returns
+        -------
+        None
+
+        """
+        self.ds = pd.DataFrame(columns=capture_elements)
+
+    def update(self, model, capture_elements):
+        """
+        Add a row to the results pandas DataFrame with the values of the
+        variables listed in capture_elements.
+
+        Parameters
+        ----------
+        model: pysd.Model
+            PySD Model object
+        capture_elements: set
+            Which model elements to capture - uses pysafe names.
+
+        Returns
+        -------
+        None
+        """
+
+        self.ds.at[model.time.round()] = [
+            getattr(model.components, key)()
+            for key in capture_elements]
+
+    def postprocess(self, **kwargs):
+        """
+        Delete time column from the pandas DataFrame and flatten xarrays if
+        required.
+
+        Returns
+        -------
+        ds: pandas.DataFrame
+            Simulation results stored as a pandas DataFrame.
+        """
+        # delete time column as it was created only for avoiding errors
+        # of appending data. See previous TODO.
+        del self.ds["time"]
+
+        # enforce flattening if df is to be saved to csv or tab file
+        flatten = True if self.out_file else kwargs.get("flatten", None)
+
+        df = DataFrameHandler.make_flat_df(
+            self.ds, kwargs["return_addresses"], flatten
+            )
+        if self.out_file:
+            self.__save_to_file(df)
+
+        return df
+
+    def __save_to_file(self, output):
+        """
+        Saves models output.
+
+        Paramters
+        ---------
+        output: pandas.DataFrame
+
+        options: argparse.Namespace
+
+        Returns
+        -------
+        None
+
+        """
+
+        if self.out_file.suffix == ".tab":
```

Comment:
@enekomartinmartinez csv exports were crashing because it was complaining that it was missing the escapechar argument. Were there any tests to check csv exports?

I tried to set escapechar="\\" and it did export, putting a '\' before the commas in column names. However, then it didn't import well.

So I opted for this approach (replacing commas in column names by '';''), which just works.
I added a couple tests to verify that csv exports now work. 

I was thinking that this change might affect users downstream, but since exporting to csv didn't seem to work anyway....

Can you give bring some light into this subject?
Thanks